docs/modules/reference/pages/daemons/daemon-config-files/pollerd.adoc

```diff
@@ -17,7 +17,8 @@ Performs xref:operation:deep-dive/service-assurance/introduction.adoc[service as
 | poller-configuration.xml
 | Contains service monitor configurations for service availability checks and global daemon configuration. +
 Restart is required when monitors are created or removed. +
-A reload event loads changed configuration parameters of existing monitors, but will not load new monitors.
+A reload event loads changed configuration parameters of existing monitors, but will not load new monitors. +
+A reload event loads changes within a package definition including `filter`, `include-range` and `exclude-range`. Outside of a package definition that is either an element or attribute of `poller-configuration`, a restart is required for the changes to take effect. 
```

Comment:
@ProjectJYL I am really sorry for being picky here :hugs: 
I think my point is still not flashed out here correctly.

> A reload event loads changed configuration parameters of existing monitors, but will not load new monitors. 
> A reload event loads changes within a package definition including `filter`, `include-range` and `exclude-range`.

Does it only reload changes in existing parameters in a monitor, or does it also reload parameters, that were not there before?
The first sentence could mean that. But also could miss that. How can you change a parameter that was not there before? "add into an existing monitor" != "change configuration parameter of existing monitor".

I think you already got my point before, but let me give you an example:

This monitor is loaded already due to a restart:
```
<service name="HTTP" interval="300000" user-defined="false" status="on">
   <parameter key="retry" value="${requisition:poller-retry|requisition:retry|detector:retry|1}" />
   <parameter key="timeout" value="${requisition:poller-timeout|requisition:timeout|detector:timeout|3000}" />
   <parameter key="port" value="${requisition:port|detector:port|80}" />
   <parameter key="url" value="${requisition:url|detector:url|/}" />
   <parameter key="rrd-repository" value="${install.share.dir}/rrd/response" />
   <parameter key="rrd-base-name" value="http" />
   <parameter key="ds-name" value="http" />
</service>
```

Now I add the parameter `response` to it. Does this count as "changing" a parameter as we define in the first sentence above? For me this is not changing. It is adding / enhancing. But maybe this is because I am not a native English speaker...

```
<service name="HTTP" interval="300000" user-defined="false" status="on">
   <parameter key="retry" value="${requisition:poller-retry|requisition:retry|detector:retry|1}" />
   <parameter key="timeout" value="${requisition:poller-timeout|requisition:timeout|detector:timeout|3000}" />
   <parameter key="port" value="${requisition:port|detector:port|80}" />
   <parameter key="url" value="${requisition:url|detector:url|/}" />
   <parameter key="rrd-repository" value="${install.share.dir}/rrd/response" />
   <parameter key="rrd-base-name" value="http" />
   <parameter key="ds-name" value="http" />
   <parameter key="response" value="200" />
</service>
```

If a reload covers this, I would probably write, to avoid confusion something like:

> A reload event loads changed configuration of existing and newly added parameters of existing monitors, but will not load new monitors. 

If not, maybe:

> A reload event loads changed configuration of existing parameters of existing monitors, but will not load new monitors.

Reply:
Good point. Unfortunately, I don't have enough knowledge to modify the first sentence which I didn't write. The ambiguity lies in whether a "new" parameter that inherits default value is considered new (non-existing). For example, [JMXMonitor](https://github.com/OpenNMS/opennms/blob/96fdb6cd195079649613464e7a217b575702fba9/docs/modules/reference/pages/service-assurance/monitors/JmxMonitor.adoc#configuration-and-use). Does `factory` considered as new as it has a default value and does `beans.<variable>` considered as new as it has no default value? In my test, if I add parameter `rrd-status` and reloaded the daemon, then it shows up as effective.
![image](https://github.com/OpenNMS/opennms/assets/19956610/89797b98-cc48-482e-a789-ab66705573df)
---
W.r.t the second sentence, in my test, I can completely remove `<include-range>` and add new `<exclude-range>` and reload the daemon, it takes effect right away. "changes" here include new and existing elements.